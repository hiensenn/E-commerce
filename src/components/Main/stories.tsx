import { Meta, StoryObj } from '@storybook/react'
import Main from '.'

export default {
  title: 'Main',
  component: Main,
  args: {
    title: 'Tile Default',
    description: 'Description Default'
  },
  parameters: {
    layout: 'fullscreen'
  }
} as Meta

export const Default: StoryObj = {}

export const Basic: StoryObj = {
  args: {
    title: 'Hiensen',
    description: 'TypeScript, ReactJS, NesxtJS e Styled-Components'
  }
}
