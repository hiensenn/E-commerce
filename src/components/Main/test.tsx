import { render, screen } from '@testing-library/react'
import { composeStories } from '@storybook/testing-react';
import * as stories from './stories';

import Main from '.'

const { Default } = composeStories(stories);

describe('<Main />', () => {
  it('should render the heading', () => {
    // renderiza o component
    const { container } = render(<Main />)

    // busca o elemento e verifica a existência dele
    expect(
      screen.getByRole('heading', { name: /React Avançado/i })
    ).toBeInTheDocument()

    // gerar snapshot
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the colors correctly', () => {
    // renderiza o component
    const { container } = render(<Main />)

    // verifica se o background-color está correto
    expect(container.firstChild).toHaveStyle({ 'background-color': '#06092b' })
  })

  it('should render stories', () => {
    // renderiza o component
    const { container } = render(<Main />)

    // verifica se o background-color está correto
    expect(container.firstChild).toHaveStyle({ 'background-color': '#06092b' })
  })

  
})

describe('Main Component Stories', () => {
  it('should render the default story correctly', () => {
    render(<Default />);

    expect(screen.getByRole('heading', { name: /React Avançado/i })).toBeInTheDocument();
    expect(
      screen.getByAltText(/Imagem de um átomo e React Avançado escrito ao lado./i)
    ).toBeInTheDocument();
  });
});