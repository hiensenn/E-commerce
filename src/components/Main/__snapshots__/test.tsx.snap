// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the heading 1`] = `
<main
  class="sc-blHHSb bZInSn"
>
  <img
    alt="Imagem de um átomo e React Avançado escrito ao lado."
    class="sc-gtLWhw jpGstZ"
    src="/img/logo.svg"
  />
  <h1
    class="sc-egkSDF rybla"
  >
    React Avançado
  </h1>
  <h2
    class="sc-fAUdSK kxCxIi"
  >
    TypeScript, ReactJS, NextJS e Styled Components
  </h2>
  <img
    alt="Um desenvolvedor de frente para uma tela com código."
    class="sc-dntaoT fomCpc"
    src="/img/hero-illustration.svg"
  />
</main>
`;
